src/goto-programs/show_goto_functions.cpp

```diff
@@ -55,22 +57,8 @@ void show_goto_functions(
 
   case ui_message_handlert::JSON_UI:
   {
-    //This only prints the list of functions
-    json_arrayt json_functions;
-    forall_goto_functions(it, goto_functions)
-    {
-      json_objectt &json_function=
-        json_functions.push_back(jsont()).make_object();
-      json_function["name"]=json_stringt(id2string(it->first));
-      json_function["isBodyAvailable"]=
-        jsont::json_boolean(it->second.body_available());
-      bool is_internal=(has_prefix(id2string(it->first), CPROVER_PREFIX) ||
-                        it->first==ID__start);
-      json_function["isInternal"]=jsont::json_boolean(is_internal);
-    }
-    json_objectt json_result;
-    json_result["functions"]=json_functions;
-    std::cout << ",\n" << json_result;
+    show_goto_functions_jsont json_show_functions(ns);
+    json_show_functions.show_goto_functions(goto_functions);
```

Comment:
I would suggest to get it done right away, yes. (Rationale: Given the high rate at which PRs are being produced at the moment, it seems timely to move to more complete PRs instead of many smaller ones.)